### ISSUE TITLE
Content should be labelable and should also return an <article>

### ISSUE THREAD

**Issue by silviuaavram** [adobe/react-spectrum#1776]
# 🙋 Feature Request

The `Content` component should be extended with the following:

1. It should accept label props. I should be able to label my content sections with an aria-label or with the heading inside the sections. 
2. The DOM element should also be an `article` if I choose to. I don't have a strong opinion on this one, but as far as I know, section and article are both important semantics when it comes to content. One is part of a larger content element, the other one is independent. I feel that we should accept both.

I am happy to implement this if we agree on the specifications. It should include tests, documentation and component changes.

## 🤔 Expected Behavior

1. The Content should be labelable.
2. The Content should accept a prop to change the section to article.

## 😯 Current Behavior

1. Label props are discarded.
2. Only `<section>` is used.

## 💁 Possible Solution

1. Easy fix in the Content component via filterDOMProps.
2. Easy fix as well but the API needs to change.

## 🔦 Context

I am trying to have an independent labelled by an ID content, which to my mind is an `article` labelled by the Heading.

## 🧢 Your Company/Team

Adobe Target.

**Comment by silviuaavram** [comment]
This ticket is relevant when using `Content` outside a `Dialog`, so if its main purpose is to be in a `Dialog` then we can close this and I could use standard HTML 5. Let me know what you think.

**Comment by snowystinger** [comment]
`Content`, `Text`, or any of the components that say this `within a Spectrum container such as a Dialog or IllustratedMessage` in their documentation are not there for standalone use. Your use case is a little interesting though, if Spectrum represents this pattern, then maybe Content should allow for `article` as well, depending on where it is used. Can you share your mocks?
